extensions/gdx-jnigen/src/com/badlogic/gdx/jnigen/NativeCodeGenerator.java

```diff
@@ -297,14 +301,41 @@ private void generateHFile (FileDescriptor file) throws Exception {
 	protected void emitHeaderInclude (StringBuffer buffer, String fileName) {
 		buffer.append("#include <" + fileName + ">\n");
 	}
+	
+	/** Emits a definition for desktop, so critical natives are only included in natives for desktop platforms */
+	private void emitDesktopDefinition (StringBuffer buffer) {

```

Comment:
It is only generated for files which contain critical natives.
So I don't think it pollutes every code generated.

I'm also fine with passing `__LIBGDX__DESKTOP__` by the initiator.